src/components/Homepage.tsx

```diff
@@ -11,12 +13,45 @@ import { PostType } from "../lib/types";
 
 const Homepage = () => {
   const navigateTo = useNavigate();
+  const [search, updateSearch] = useState("");
   const [query, updateQuery] = useState("");
```

Comment:
This is an interesting design. However, I think `query` and `search` can be merged as one, where updates directly shows up in the query box.

That being said, we can offer user a separate textbox in the advanced search options.

Think of the "all these words: " box in [Google's Advanced Search page](https://www.google.com/advanced_search).